lib/NMISNG/RPC.pm

```diff
@@ -0,0 +1,82 @@
+package NMISNG::RPC;
+
+our $VERSION = "9.4.8";
+
+use Mojo::Base -base;
+use Carp;
+use JSON::XS;
+use Mojo::UserAgent;
+
+#Wraps Mojo user agent to make JSON-RPC 2.0 requests, inspired by MojoX::JSONRPC2::HTTP;
+#We will need a long requesttime out for large snmp requests
+use constant REQUEST_TIMEOUT    => 300;
+has ua      => sub {
+    Mojo::UserAgent->new
+        ->inactivity_timeout(0)
+        ->request_timeout(REQUEST_TIMEOUT)
+};
+#TODO make this configurable   
+has url => 'http://localhost:9000/rpc';
+
+
+sub call  { return shift->_request('call',@_) }
+
+sub _request {
+    my ($self, $func, $method, @params) = @_;
+
+    #Make sure we have params to pass or set to empty array
+    @params = () unless @params;
+
+    #TODO fix this for async
+    my $id = 1;
+    #Make a post request with our JSON-RPC payload
+    #No notify yet just a simple request
+
+    my $json = {
+        jsonrpc => '2.0',
+        method => $method,
+        params => \@params,
+        id => $id,
+    };
+
+    use Data::Dumper;
+    print(Dumper($json));
+
+    my $tx = $self->ua->post(
```

Comment:
I think we need to handle the die here as well, from the docs:

# Fine grained response handling (dies on connection errors)